resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/undo/snapshot/IContainerSnapshot.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Red Hat Inc - Adapted from classes in org.eclipse.ui.ide.undo and org.eclipse.ui.internal.ide.undo
+ *******************************************************************************/
+package org.eclipse.core.resources.undo.snapshot;
+
+import java.net.URI;
+import org.eclipse.core.resources.IResourceFilterDescription;
+
+/**
+ * IContainerSnapshot is a lightweight description that describes a container to
+ * be created.
+ *
+ * This class is not intended to be instantiated by clients.
```

Comment:
This line can be removed as interfaces cannot be instantiated.
However, 1 important thing to settle when introducing this interface API is whether we want to allow other implementations? If we want to restrict the implementations to the ones that are shipped here, we should add a `@noimplements` tag to make PDE warn in case someone wants to implement it.
It's safer to start with as many restrictions as possible and to remove them later than the other way round, as it gives more freedom for further changes.